src/libmongoc/src/mongoc/mongoc-topology-private.h

```diff
@@ -95,14 +121,20 @@ typedef struct _mongoc_topology_t {
    bson_thread_t srv_polling_thread;
    mongoc_cond_t srv_polling_cond;
 
-   bson_mutex_t mutex;
```

Comment:
Looks like there's a bit more work to do in the case of CSE. I'll have to take a deeper look here, since there may be more shared state that I didn't notice.